flycheck.el

```diff
@@ -3072,7 +3084,7 @@ nil."
                      "")))
                 (`interrupted "-")
                 (`suspicious "?"))))
-    (concat " FlyC" text)))
+    (concat " " flycheck-mode-line-lighter-base text)))
```

Comment:
I'd leave it as is and always add the space.  The option is for the sake of simplicity, and should be fail-safe, in my opinion.  

Those who'd like to remove the space as well should just customise `flycheck-mode-line` directly.  If the space is that much pain you'll probably have gone to much greater length to shrink your mode line anyway, I think.